src/device.ts

```diff
@@ -0,0 +1,56 @@
+import { EventEmitter } from "events";
+import { Hub } from "./hub";
+
+import * as Consts from "./consts";
+
+export class Device extends EventEmitter {
+
+    private _hub: Hub;
+    private _portId: number;
+    private _connected: boolean = true;
+    private _type: number;
+    private _busy: boolean = false;
+    private _finished: (() => void) | null = null;
+
+    constructor (hub: Hub, portId: number, type: number = Consts.DeviceType.UNKNOWN) {
+        super();
+        this._hub = hub;
+        this._portId = portId;
+        this._type = type;
+        this.hub.on("detach", (device) => {
+            if (device.portId === this.portId) {
```

Comment:
Wouldn't `device === this` be better? This way old disconnected devices would not generate `detach` events.

Reply:
Good catch. Will change this.

Reply:
Maybe a more straightforward approach would be to remove this listener on detach.

Reply:
@nutki Listener is now removed on detach.